config/gulp.js

```diff
@@ -1,103 +1,117 @@
-'use strict'
+'use strict';
 
-const gulp = require('gulp')
-const babel = require('gulp-babel')
-const sass = require('gulp-sass')
-const webpack = require('webpack-stream')
-const watch = require('gulp-watch')
-const path = require('path')
+const gulp = require('gulp');
+const babel = require('gulp-babel');
+const sass = require('gulp-sass');
+const webpack = require('webpack-stream');
+const path = require('path');
 
 const babelConfig = {
-  presets: ['react', 'es2015'],
-  plugins: [
-    [
-      'css-modules-transform', {
-        preprocessCss: 'preprocessor.js',
-        generateScopedName: '[name]__[local]___[hash:base64:5]',
-        extensions: ['.css', '.scss'],
-        extractCss: './dist/styles/production.css'
-      }
+    presets: ['react', 'es2015'],
+    plugins: [
+        [
+            'css-modules-transform', {
+                preprocessCss: 'preprocessor.js',
+                generateScopedName: '[name]__[local]___[hash:base64:5]',
+                extensions: ['.css', '.scss'],
+                extractCss: './dist/styles/production.css'
+            }
+        ]
     ]
-  ]
-}
+};
 
 module.exports = {
 
-  defaultTaskName: 'default',
+    defaultTaskName: 'default',
 
-  tasks: {
+    tasks: {
 
-    default: ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient', 'watch'],
+        default: ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient', 'watchTemplates', 'watchStyles', 'watchImages'],
 
-    watch: () => {
-      return gulp.watch(
-        path.join('frontend', '**', '*'),
-        ['compileTemplate', 'compileStyles', 'moveImages', 'bundleAdmin', 'bundleClient'])
-    },
+        watchTemplates: () => {
+            return gulp.watch(
+                path.join('frontend', 'js', '**', '*'),
+                ['compileTemplate', 'bundleAdmin', 'bundleClient']
```

Comment:
Not really. This watcher is looking for changes in the JS (react templates, redux) and is responsible for rebuilding templates, rebuilding the admin and client bundles.  It could probably be split further into tasks for compiling the frontend and admin panel resources.